[ICE]: `no entry for node id`

<!--
[31mICE[0m: Rustc ./a.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_ast_lowering/src/lib.rs:684:13: 'no entry for node id: `NodeId(639)`'', 'thread 'rustc' ($TID) panicked at compiler/rustc_ast_lowering/src/lib.rs:684:13: 'no entry for node id: `NodeId(639)`''
File: /tmp/im6/a.rs
-->

auto-reduced (treereduce-rust):
````rust
trait X {}

impl X {
    reuse Iterator::fold {
		let _: &X;
    }
}

````

original:
````rust
#![feature(fn_delegation)]
trait X {}

impl X {
    reuse Iterator::fold {
		let _: &X;
    }
}

````

Version information
````
rustc 1.97.0-nightly (b954122bb 2026-05-20)
binary: rustc
commit-hash: b954122bbe6c4a244c511a9a168c6d5438d8f58f
commit-date: 2026-05-20
host: x86_64-unknown-linux-gnu
release: 1.97.0-nightly
LLVM version: 22.1.4

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/b954122bbe6c4a244c511a9a168c6d5438d8f58f/compiler/rustc_ast_lowering/src/lib.rs#L678-L690

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0658]: functions delegation is not yet fully implemented
 --> /tmp/icemaker_global_tempdir.i5tRtAV11AVg/rustc_testrunner_tmpdir_reporting.q39cQrbGUCpM/mvce.rs:4:5
  |
4 | /     reuse Iterator::fold {
5 | |         let _: &X;
6 | |     }
  | |_____^
  |
  = note: see issue #118212 <https://github.com/rust-lang/rust/issues/118212> for more information
  = help: add `#![feature(fn_delegation)]` to the crate attributes to enable
  = note: this compiler was built on 2026-05-20; consider upgrading it if it is out of date


thread 'rustc' (1032756) panicked at /rustc-dev/b954122bbe6c4a244c511a9a168c6d5438d8f58f/compiler/rustc_ast_lowering/src/lib.rs:684:13:
no entry for node id: `NodeId(24)`
stack backtrace:
   0:     0x7ff5ce89afb9 - <<std[9b1fefb192d7dcdd]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[8f1d9dc20add8ccc]::fmt::Display>::fmt
   1:     0x7ff5cf01fc88 - core[8f1d9dc20add8ccc]::fmt::write
   2:     0x7ff5ce8b1ca6 - <std[9b1fefb192d7dcdd]::sys::stdio::unix::Stderr as std[9b1fefb192d7dcdd]::io::Write>::write_fmt
   3:     0x7ff5ce8711de - std[9b1fefb192d7dcdd]::panicking::default_hook::{closure#0}
   4:     0x7ff5ce88e8f3 - std[9b1fefb192d7dcdd]::panicking::default_hook
   5:     0x7ff5cd68d5a1 - std[9b1fefb192d7dcdd]::panicking::update_hook::<alloc[cf25b8836bdc34f1]::boxed::Box<rustc_driver_impl[95c61dc90027d8c1]::install_ice_hook::{closure#1}>>::{closure#0}
   6:     0x7ff5ce88ebd2 - std[9b1fefb192d7dcdd]::panicking::panic_with_hook
   7:     0x7ff5ce8712a2 - std[9b1fefb192d7dcdd]::panicking::panic_handler::{closure#0}
   8:     0x7ff5ce8657c9 - std[9b1fefb192d7dcdd]::sys::backtrace::__rust_end_short_backtrace::<std[9b1fefb192d7dcdd]::panicking::panic_handler::{closure#0}, !>
   9:     0x7ff5ce872d0d - __rustc[643a9a89e4a1a7a2]::rust_begin_unwind
  10:     0x7ff5cb19438c - core[8f1d9dc20add8ccc]::panicking::panic_fmt
  11:     0x7ff5cf1d2b10 - <rustc_ast_lowering[6fb7a663fca20012]::LoweringContext>::new_named_lifetime
  12:     0x7ff5cf1d3dd1 - <rustc_ast_lowering[6fb7a663fca20012]::LoweringContext>::lower_ty
  13:     0x7ff5cf750817 - <rustc_ast_lowering[6fb7a663fca20012]::LoweringContext>::lower_stmts
  14:     0x7ff5cf77b306 - <rustc_ast_lowering[6fb7a663fca20012]::LoweringContext>::lower_block
  15:     0x7ff5cd1812a2 - <rustc_ast_lowering[6fb7a663fca20012]::LoweringContext>::lower_target_expr
  16:     0x7ff5cd17dae4 - <rustc_ast_lowering[6fb7a663fca20012]::LoweringContext>::lower_delegation
  17:     0x7ff5cfa471aa - <rustc_ast_lowering[6fb7a663fca20012]::item::ItemLowerer>::lower_node
  18:     0x7ff5cd17be4b - rustc_ast_lowering[6fb7a663fca20012]::lower_delayed_owner
  19:     0x7ff5cf2a57ae - rustc_query_impl[2fa820134a6ba49a]::execution::try_execute_query::<rustc_data_structures[ea7f2117d5faad46]::vec_cache::VecCache<rustc_span[5bdacda662c6a8f1]::def_id::LocalDefId, rustc_middle[dee0145e95396fd4]::query::erase::ErasedData<[u8; 0usize]>, rustc_middle[dee0145e95396fd4]::dep_graph::graph::DepNodeIndex>, false>
  20:     0x7ff5ce27bb6b - rustc_query_impl[2fa820134a6ba49a]::query_impl::lower_delayed_owner::execute_query_non_incr::__rust_end_short_backtrace
  21:     0x7ff5d032286c - rustc_middle[dee0145e95396fd4]::hir::map::hir_crate_items
  22:     0x7ff5d01a11d9 - rustc_query_impl[2fa820134a6ba49a]::query_impl::hir_crate_items::invoke_provider_fn::__rust_begin_short_backtrace
  23:     0x7ff5d01a5c71 - rustc_query_impl[2fa820134a6ba49a]::execution::try_execute_query::<rustc_middle[dee0145e95396fd4]::query::caches::SingleCache<rustc_middle[dee0145e95396fd4]::query::erase::ErasedData<[u8; 8usize]>>, false>
  24:     0x7ff5d01a6f43 - rustc_query_impl[2fa820134a6ba49a]::query_impl::hir_crate_items::execute_query_non_incr::__rust_end_short_backtrace
  25:     0x7ff5cf2a2dd8 - rustc_interface[f0cd769d5a742462]::passes::analysis
  26:     0x7ff5d01a6849 - rustc_query_impl[2fa820134a6ba49a]::execution::try_execute_query::<rustc_middle[dee0145e95396fd4]::query::caches::SingleCache<rustc_middle[dee0145e95396fd4]::query::erase::ErasedData<[u8; 0usize]>>, false>
  27:     0x7ff5d01a6529 - rustc_query_impl[2fa820134a6ba49a]::query_impl::analysis::execute_query_non_incr::__rust_end_short_backtrace
  28:     0x7ff5d0157e5b - rustc_interface[f0cd769d5a742462]::interface::run_compiler::<(), rustc_driver_impl[95c61dc90027d8c1]::run_compiler::{closure#0}>::{closure#1}
  29:     0x7ff5d01910c0 - std[9b1fefb192d7dcdd]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[f0cd769d5a742462]::util::run_in_thread_with_globals<rustc_interface[f0cd769d5a742462]::util::run_in_thread_pool_with_globals<rustc_interface[f0cd769d5a742462]::interface::run_compiler<(), rustc_driver_impl[95c61dc90027d8c1]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  30:     0x7ff5d01917ad - <std[9b1fefb192d7dcdd]::thread::lifecycle::spawn_unchecked<rustc_interface[f0cd769d5a742462]::util::run_in_thread_with_globals<rustc_interface[f0cd769d5a742462]::util::run_in_thread_pool_with_globals<rustc_interface[f0cd769d5a742462]::interface::run_compiler<(), rustc_driver_impl[95c61dc90027d8c1]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[8f1d9dc20add8ccc]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  31:     0x7ff5d01925ac - <std[9b1fefb192d7dcdd]::sys::thread::unix::Thread>::new::thread_start
  32:     0x7ff5c96981b9 - <unknown>
  33:     0x7ff5c971d21c - <unknown>
  34:                0x0 - <unknown>

error: the compiler unexpectedly panicked. This is a bug

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.97.0-nightly (b954122bb 2026-05-20) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [lower_delayed_owner] lowering the delayed AST owner `<impl at /tmp/icemaker_global_tempdir.i5tRtAV11AVg/rustc_testrunner_tmpdir_reporting.q39cQrbGUCpM/mvce.rs:3:1: 3:7>::fold`
#1 [hir_crate_items] getting HIR crate items
#2 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0658`.

```

</p>
</details>

<!--
query stack:
#0 [lower_delayed_owner] lowering the delayed AST owner `<impl at /tmp/icemaker_global_tempdir.i5tRtAV11AVg/rustc_testrunner_tmpdir_reporting.q39cQrbGUCpM/mvce.rs:3:1: 3:7>::fold`
#1 [hir_crate_items] getting HIR crate items
#2 [analysis] running analysis passes on crate `mvce`

-->
@rustbot label +F-fn_delegation

--- a/compiler/rustc_resolve/src/late.rs
+++ b/compiler/rustc_resolve/src/late.rs
@@ -3937,7 +3937,9 @@ impl<'a, 'ast, 'ra, 'tcx> LateResolutionVisitor<'a, 'ast, 'ra, 'tcx> {
 
             //As we lower target_expr_template body to a body of a function we need a label rib (#148889)
             this.with_label_rib(RibKind::FnOrCoroutine, |this| {
-                this.visit_block(body);
+                this.with_lifetime_rib(LifetimeRibKind::Elided(LifetimeRes::Infer), |this| {
+                    this.visit_block(body);
+                });
             });
         });
     }

--- a/tests/ui/delegation/ice-line-bounds-issue-148732.rs
+++ b/tests/ui/delegation/ice-line-bounds-issue-148732.rs
@@ -2,7 +2,7 @@ reuse a as b {
     //~^ ERROR cannot find function `a` in this scope
     //~| ERROR functions delegation is not yet fully implemented
     dbg!(b);
-    //~^ ERROR missing lifetime specifier
+    //~^ ERROR: `fn() {b}` doesn't implement `Debug`
 }
 
 fn main() {}

--- a/tests/ui/delegation/ice-line-bounds-issue-148732.stderr
+++ b/tests/ui/delegation/ice-line-bounds-issue-148732.stderr
@@ -1,9 +1,3 @@
-error[E0106]: missing lifetime specifier
-  --> $DIR/ice-line-bounds-issue-148732.rs:4:5
-   |
-LL |     dbg!(b);
-   |     ^^^^^^^ expected named lifetime parameter
-
 error[E0425]: cannot find function `a` in this scope
   --> $DIR/ice-line-bounds-issue-148732.rs:1:7
    |
@@ -25,7 +19,18 @@ LL | | }
    = help: add `#![feature(fn_delegation)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
 
+error[E0277]: `fn() {b}` doesn't implement `Debug`
+  --> $DIR/ice-line-bounds-issue-148732.rs:4:5
+   |
+LL | reuse a as b {
+   |            - consider calling this function
+...
+LL |     dbg!(b);
+   |     ^^^^^^^ the trait `Debug` is not implemented for fn item `fn() {b}`
+   |
+   = help: use parentheses to call this function: `b()`
+
 error: aborting due to 3 previous errors
 
-Some errors have detailed explanations: E0106, E0425, E0658.
-For more information about an error, try `rustc --explain E0106`.
+Some errors have detailed explanations: E0277, E0425, E0658.
+For more information about an error, try `rustc --explain E0277`.

--- a/tests/ui/delegation/wrong-lifetime-rib.rs
+++ b/tests/ui/delegation/wrong-lifetime-rib.rs
@@ -29,4 +29,14 @@ mod ice_156758 {
     }
 }
 
+mod ice_156806 {
+    trait X {}
+
+    impl X { //~ ERROR: expected a type, found a trait
+        reuse Iterator::fold { //~ ERROR: `()` is not an iterator
+            let _: &X; //~ ERROR: expected a type, found a trait
+        }
+    }
+}
+
 fn main() {}

--- a/tests/ui/delegation/wrong-lifetime-rib.stderr
+++ b/tests/ui/delegation/wrong-lifetime-rib.stderr
@@ -19,6 +19,21 @@ help: you might have intended to implement this trait for a given type
 LL |     impl X for /* Type */ {
    |            ++++++++++++++
 
+error[E0782]: expected a type, found a trait
+  --> $DIR/wrong-lifetime-rib.rs:35:10
+   |
+LL |     impl X {
+   |          ^
+   |
+help: you can add the `dyn` keyword if you want a trait object
+   |
+LL |     impl dyn X {
+   |          +++
+help: you might have intended to implement this trait for a given type
+   |
+LL |     impl X for /* Type */ {
+   |            ++++++++++++++
+
 error[E0116]: cannot define inherent `impl` for a type outside of the crate where the type is defined
   --> $DIR/wrong-lifetime-rib.rs:9:5
    |
@@ -40,7 +55,28 @@ LL -         reuse<<<&Project> :: Ty> :: Ty as Iterator>::next;
 LL +         reuse<<<&() as Example>::Ty> :: Ty as Iterator>::next;
    |
 
-error: aborting due to 4 previous errors
+error[E0782]: expected a type, found a trait
+  --> $DIR/wrong-lifetime-rib.rs:37:21
+   |
+LL |             let _: &X;
+   |                     ^
+   |
+help: you can add the `dyn` keyword if you want a trait object
+   |
+LL |             let _: &dyn X;
+   |                     +++
 
-Some errors have detailed explanations: E0116, E0223, E0423, E0782.
+error[E0599]: `()` is not an iterator
+  --> $DIR/wrong-lifetime-rib.rs:36:25
+   |
+LL |         reuse Iterator::fold {
+   |                         ^^^^ `()` is not an iterator
+   |
+   = note: the following trait bounds were not satisfied:
+           `(): Iterator`
+           which is required by `&mut (): Iterator`
+
+error: aborting due to 7 previous errors
+
+Some errors have detailed explanations: E0116, E0223, E0423, E0599, E0782.
 For more information about an error, try `rustc --explain E0116`.

